crates/iota-light-client/src/config.rs

```diff
@@ -0,0 +1,211 @@
+    /// A config to sync the light client from an archive store. Since the
+    /// archive does not store full checkpoints, it cannot be used to
+    /// check objects/transactions.
+    pub archive_store_config: Option<ObjectStoreConfig>,
+}
+
+impl Config {
+    pub fn get_mainnet_config() -> Self {
+        Self::get_config_by_network("mainnet")
+    }
+
+    pub fn get_testnet_config() -> Self {
+        Self::get_config_by_network("testnet")
+    }
+
+    pub fn get_devnet_config() -> Self {
+        Self::get_config_by_network("devnet")
+    }
+
+    fn get_config_by_network(network: &str) -> Self {
+        Self {
+            rpc_url: Url::parse(&format!("https://api.{network}.iota.cafe")).unwrap(),
+            graphql_url: Some(Url::parse(&format!("https://graphql.{network}.iota.cafe")).unwrap()),
+            checkpoints_dir: PathBuf::from_str(&format!("checkpoints_{network}")).unwrap(),
+            genesis_blob_download_url: Some(
+                Url::parse(&format!("https://dbfiles.{network}.iota.cafe/genesis.blob")).unwrap(),
+            ),
+            sync_before_check: false,
+            checkpoint_store_config: Some(ObjectStoreConfig {
+                object_store: Some(ObjectStoreType::S3),
+                object_store_connection_limit: 20,
+                aws_endpoint: Some(format!("https://checkpoints.{network}.iota.cafe")),
+                aws_virtual_hosted_style_request: true,
+                aws_region: Some("weur".to_string()),
+                no_sign_request: true,
+                ..Default::default()
+            }),
+            archive_store_config: Some(ObjectStoreConfig {
+                object_store: Some(ObjectStoreType::S3),
+                object_store_connection_limit: 20,
+                aws_endpoint: Some(format!("https://archive.{network}.iota.cafe")),
+                aws_virtual_hosted_style_request: true,
+                aws_region: Some("weur".to_string()),
+                no_sign_request: true,
+                ..Default::default()
+            }),
+        }
+    }
+}
+
+impl Config {
+    /// Loads the config from file.
+    pub async fn load<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let content = read_to_string(path).await?;
+        let config: Config = serde_yaml::from_str(&content)?;
+        config.validate()?;
+        Ok(config)
+    }
+
+    /// Creates the necessary checkpoint directory and files if not already
+    /// present.
+    pub async fn setup(&self) -> Result<()> {
+        // Create the checkpoints directory if it doesn't exist yet
+        if !self.checkpoints_dir.is_dir() {
+            create_dir_all(&self.checkpoints_dir).await?;
+        }
+        // Download or copy the genesis blob if it doesn't exist yet
+        if !self.genesis_blob_file_path().is_file() {
+            if let Some(url) = &self.genesis_blob_download_url {
+                match url.scheme() {
+                    "file" => {
```

Comment:
What does that mean? Local URL to a file?

Reply:
I mean a URL of the format: `file://<some-path-to-a-file>`

Reply:
it's useful feature for localnets, so you don't have to manually copy over the `genesis.blob`